src/Algebra/Properties/Ring.agda

```diff
@@ -67,3 +69,23 @@ open AbelianGroupProperties +-abelianGroup public
   - (1# * x)  ≈⟨ -‿cong ( *-identityˡ x ) ⟩
   - x         ∎
 
+x≈0 : ∀ x → x + x ≈ x → x ≈ 0#
+x≈0 x eq = begin
+  x            ≈⟨ sym(+-identityʳ x) ⟩
+  x + 0#       ≈⟨ +-congˡ (sym (-‿inverseʳ x)) ⟩
+  x + (x - x)  ≈⟨ sym (+-assoc x x (- x)) ⟩
+  x + x - x    ≈⟨ +-congʳ(eq) ⟩
+  x - x        ≈⟨ -‿inverseʳ x ⟩
+  0#           ∎
+
+xy-z≈xy-xz : ∀ x y z → x * (y - z) ≈ x * y - x * z
```

Comment:
Definitely need some brackets in these names, e.g. `x[y-z]≈xy-xz`. Actually aren't these just distributivity proofs and therefore should use the `Distributive` types + names as appropriate?

Reply:
Hmm I have added brackets. We have `-‿distribˡ-*`, may be `+-distrib-*` ? Let me know if you have better names.